src/app/modules/add-user/add-user.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react';
+
+import { useMutation } from '@apollo/client';
+import { format } from 'date-fns';
+
+import { PageAddUser } from '../../../atomic/pag.add-user/add-user.component';
+import { Mutation } from '../../data/graphql/graphql.schemas';
+import { Regex } from '../../utils/regex';
+
+interface AddUserVariables {
+  data: { name: string; email: string; phone: string; birthDate: string; password: string; role: string };
```

Comment:
A tipagem do `role` poderia estar melhor especificada do que apenas uma string, o que acha?

Reply:
Então eu penso que se a aplicação crescer teria que alterar vários arquivos. Se for só string eu só altero no Selector

Reply:
Aplicação crescer = mais roles acrescentadas?

Nesse caso vc só acrescentaria as opções adicionais no tipo, não? Não vale a pena essa pequena mudança a mais pra garantir a consistência do seletor?